### DISCUSSION TITLE
一个关于 fcitx5-lua 插件的问题

### DISCUSSION THREAD

**Discussion by ClarkQAQ** [fcitx/fcitx5#671] [Q&A] [answered]
```lua
function QuickPhrase(text)
    print("quickPhrase:" .. text)
    if text == "test" then
        print("匹配成功:" .. text)
        return "测试"
    end
end

local function registerModule()
    fcitx.addQuickPhraseHandler("QuickPhrase")
end
```

在如上代码中使用快速输入功能输入 `test` 后, 我应该如何正确的将 `测试` 传入候选
目前输入会打印匹配成功但是候选没有 return 的内容 `测试`

**Comment by ClarkQAQ** [comment]
得了.....看起来只有 `commitString` 直接上屏这种方式......

**Comment by wengxt** [comment, marked as answer]
QuickPhrase 的 handler 的返回值必须符合特殊的要求，并不是返回字符串。你的返回结果不被认可。

首先返回结果是一个 table ，其次返回的每一项都需要是一个三元组
{ 显示 , 提交, 类型 }

例如
 {"A", "B", fcitx.QuickPhraseAction.Commit} 表示显示成 A ，提交成 B，类型是直接提交的候选词

其他还有一些特殊的类型，例如 fcitx.QuickPhraseAction.TypeToBuffer ，表示把候选继续输入到 quickphrase 的缓冲区中

fcitx.QuickPhraseAction.Break ，避免后续其他的 handler 处理

还有一些其他特殊的值的可以改变候选词按键类型之类，可以参考

https://github.com/fcitx/fcitx5-lua/blob/master/src/imeapi/imeapi.lua#L58

**Reply by ClarkQAQ** [reply]
感谢回复, 他现在是工作的!!!
